### PR TITLE
Update demo document for Lorenz with exogenous input. inverse

### PR DESCRIPTION
**Pull request by Jerry-Jzy**
## what is the purpose of the change
Update demo document for Lorenz with exogenous input. inverse

### CODE REVIEW

**Comment by lululxvi** [comment]
Use file name: Lorenz.inverse.forced.rst

**Comment by Jerry-Jzy** [comment]
> Use file name: Lorenz.inverse.forced.rst
Gotcha

File: docs/demos/pinn_inverse.rst [hunk 1]
```diff
@@ -10,8 +10,8 @@ ODEs
    :maxdepth: 1
 
    pinn_inverse/lorenz.inverse
+   pinn_inverse/lorenz_with_exogenous_input.inverse
```
Comment (lululxvi) [inline docs/demos/pinn_inverse.rst:14]: Delete one empty line

Comment (Jerry-Jzy) [reply docs/demos/pinn_inverse.rst:14]: any other problems?

Comment (lululxvi) [reply docs/demos/pinn_inverse.rst:14]: Solve these first.

Comment (Jerry-Jzy) [reply docs/demos/pinn_inverse.rst]: Already fixed.

**Comment by lululxvi** [comment]
Cannot see it at https://deepxde--709.org.readthedocs.build/en/709/demos/pinn_inverse.html

**Comment by Jerry-Jzy** [comment]
> Cannot see it at https://deepxde--709.org.readthedocs.build/en/709/demos/pinn_inverse.html

fixed, can see it now.

**Comment by lululxvi** [comment]
Use file name: lorenz.inverse.forced.rst --- small "l"

File: docs/demos/pinn_inverse/Lorenz.inverse.forced.rst [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    maxtime = 3
+    time = np.linspace(0, maxtime, 200)
+    ex_input = 10 * np.sin(2 * np.pi * time)
+
+Next, we can define the interpolation function of time and exogenous input:
+
+.. code-block:: python
+
+    def ex_func2(t):
+    spline = sp.interpolate.Rbf(
```
Comment (lululxvi) [inline docs/demos/pinn_inverse/Lorenz.inverse.forced.rst:57]: Is the format correct?

File: docs/demos/pinn_inverse/Lorenz.inverse.forced.rst [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        time, ex_input, function="thin_plate", smooth=0, episilon=0
+    )
+    return spline(t[:, 0:])
+
+Next, we create the Lorenz system to solve using the ``dde.grad.jacobian`` function.
+
+.. code-block:: python
+
+    def Lorenz_system(x, y, ex):
+    y1, y2, y3 = y[:, 0:1], y[:, 1:2], y[:, 2:]
```
Comment (lululxvi) [inline docs/demos/pinn_inverse/Lorenz.inverse.forced.rst:67]: Is the format correct?

File: docs/demos/pinn_inverse/Lorenz.inverse.forced.rst (hunk 2 above)
Comment (Jerry-Jzy) [reply docs/demos/pinn_inverse/Lorenz.inverse.forced.rst]: Fixed

File: docs/demos/pinn_inverse/Lorenz.inverse.forced.rst (hunk 3 above)
Comment (Jerry-Jzy) [reply docs/demos/pinn_inverse/Lorenz.inverse.forced.rst]: Fixed

File: docs/demos/pinn_inverse/lorenz.inverse.forced.rst [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        spline = sp.interpolate.Rbf(
+            time, ex_input, function="thin_plate", smooth=0, episilon=0
+        )
+        return spline(t)
+
+Next, we create the Lorenz system to solve using the ``dde.grad.jacobian`` function.
+
+.. code-block:: python
+
+    def LorezODE(x, t):
```
Comment (lululxvi) [inline docs/demos/pinn_inverse/lorenz.inverse.forced.rst:66]: This is not the function to define the ODE. It is `Lorenz_system`. It seems that you don't understand the code well.

Comment (Jerry-Jzy) [reply docs/demos/pinn_inverse/lorenz.inverse.forced.rst:66]: I know this part is the true solution, I just copied worry code:(

File: docs/demos/pinn_inverse/lorenz.inverse.forced.rst [hunk 5]
```diff
@@ -0,0 +1,148 @@
+Inverse problem for the Lorenz system with exogenous input
+==========================================================
+
+Problem setup
+-------------
+
+We will solve the Lorenz system:
+
+.. math:: \frac{dx}{dt} = \sigma(y-x), \quad \frac{dy}{dt} = x (\rho - z) - y, \quad \frac{dz}{dt} = x y - \beta z - 10 \sin(2\pi t) \qquad t \in [0, 3]
```
Comment (lululxvi) [inline docs/demos/pinn_inverse/lorenz.inverse.forced.rst:9]:
```x y - \beta z - f(t)

where 

f(t) = 10 \sin(2\pi t)
```

File: docs/demos/pinn_inverse/lorenz.inverse.forced.rst [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+    C2 = dde.Variable(1.0)
+    C3 = dde.Variable(1.0)
+
+Now we can begin by creating a ``TimeDomain`` class.
+
+.. code-block:: python
+    
+    geom = dde.geometry.TimeDomain(0, 3)
+
+Next, we define the exogenous input:
```
Comment (lululxvi) [inline docs/demos/pinn_inverse/lorenz.inverse.forced.rst:44]: Next, we define the exogenous input f(t):

File: docs/demos/pinn_inverse/lorenz.inverse.forced.rst (hunk 5 above)
Comment (Jerry-Jzy) [reply docs/demos/pinn_inverse/lorenz.inverse.forced.rst]: Fixed

File: docs/demos/pinn_inverse/lorenz.inverse.forced.rst (hunk 6 above)
Comment (Jerry-Jzy) [reply docs/demos/pinn_inverse/lorenz.inverse.forced.rst]: Fixed

File: docs/demos/pinn_inverse/lorenz.inverse.forced.rst [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+
+.. math:: \frac{dx}{dt} = \sigma(y-x), \quad \frac{dy}{dt} = x (\rho - z) - y, \quad \frac{dz}{dt} = x y - \beta z - f(t) \qquad t \in [0, 3]
+
+Where :math:`f(t)=10\sin(2\pi t)` is the exgoenous input.
+
+with initial conditions 
+
+.. math:: x(0) = -8, \quad y(0) = 7, \quad z(0) = 27.
+
+where the parameters :math:`\sigma`, :math:`\rho`, and :math:`\beta` are to be identified from observations of the system at certain times and whose true values are 10, 15, and 8/3, respectivly. And the :math:`\sin(2\pi t)` is the exogenous input.
```
Comment (lululxvi) [inline docs/demos/pinn_inverse/lorenz.inverse.forced.rst:17]: the f(t) is the exogenous input.

File: docs/demos/pinn_inverse/lorenz.inverse.forced.rst [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+==========================================================
+
+Problem setup
+-------------
+
+We will solve the Lorenz system:
+
+.. math:: \frac{dx}{dt} = \sigma(y-x), \quad \frac{dy}{dt} = x (\rho - z) - y, \quad \frac{dz}{dt} = x y - \beta z - f(t) \qquad t \in [0, 3]
+
+Where :math:`f(t)=10\sin(2\pi t)` is the exgoenous input.
```
Comment (lululxvi) [inline docs/demos/pinn_inverse/lorenz.inverse.forced.rst:11]: where

File: docs/demos/pinn_inverse/lorenz.inverse.forced.rst [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+
+with initial conditions
```
Comment (lululxvi) [inline docs/demos/pinn_inverse/lorenz.inverse.forced.rst:13]: The initial condition is

File: docs/demos/pinn_inverse/lorenz.inverse.forced.rst [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+    C2 = dde.Variable(1.0)
+    C3 = dde.Variable(1.0)
+
+Now we can begin by creating a ``TimeDomain`` class.
+
+.. code-block:: python
+    
+    geom = dde.geometry.TimeDomain(0, 3)
+
+Next, we define the exogenous input :math:`f(t)`:
```
Comment (lululxvi) [inline docs/demos/pinn_inverse/lorenz.inverse.forced.rst:46]: We assume that we don't know the formula of f(t), and we only know f(t) at 200 points.

File: docs/demos/pinn_inverse/lorenz.inverse.forced.rst [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+
+.. code-block:: python
+
+    maxtime = 3
+    time = np.linspace(0, maxtime, 200)
+    ex_input = 10 * np.sin(2 * np.pi * time)
+
+Next, we can define the interpolation function of time and exogenous input:
```
Comment (lululxvi) [inline docs/demos/pinn_inverse/lorenz.inverse.forced.rst:54]: Next, we can define an interpolation function for f(t):

File: docs/demos/pinn_inverse/lorenz.inverse.forced.rst [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+        geom,
+        Lorenz_system,
+        [ic1, ic2, ic3, observe_y0, observe_y1, observe_y2],
+        num_domain=400,
+        num_boundary=2,
+        anchors=observe_t,
+        auxiliary_var_function=ex_func2,
+    )
+
+Where ``num_domain`` is the number of points inside the domain, and ``num_boundary`` is the number of points on the boundary. ``anchors`` are extra points beyond ``num_domain`` and ``num_boundary`` used for training. ``auxiliary_var_function`` is the interpolation function of time and exogenous input we defined above.
```
Comment (lululxvi) [inline docs/demos/pinn_inverse/lorenz.inverse.forced.rst:119]: is the interpolation function of f(t) we defined above.

**Pull request merged**